scripts/lib/api/generateMdxComponents.ts

```diff
@@ -0,0 +1,361 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2024.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import { CheerioAPI, Cheerio, Element } from "cheerio";
+import { unified } from "unified";
+import rehypeParse from "rehype-parse";
+import rehypeRemark from "rehype-remark";
+import remarkStringify from "remark-stringify";
+
+import { ApiType } from "./Metadata";
+import { getLastPartFromFullIdentifier } from "../stringUtils";
+
+export type componentProps = {
+  id: string;
+  name?: string;
+  attributeType?: string;
+  attributeValue?: string;
+  githubSourceLink?: string;
+  signature?: string;
+  extraSignatures?: string[];
+};
+
+export async function processMdxComponent(
+  $: CheerioAPI,
+  $main: Cheerio<any>,
+  signatures: Cheerio<Element>[],
+  $dl: Cheerio<any>,
+  priorApiType: ApiType | undefined,
+  apiType: ApiType,
+  id: string,
+): Promise<[string, string]> {
+  findByText($, $main, "em.property", apiType).remove();
+
+  const $firstSignature = signatures.shift()!;
+  const githubSourceLink = prepareGitHubLink(
+    $firstSignature,
+    apiType === "method",
+  );
+
+  const componentProps = prepareProps(
+    $,
+    $firstSignature,
+    $dl,
+    priorApiType,
```

Comment:
This argument needs to be different for overload signatures (`apiType`) than for the first call (`priorApiType`)